File: dotnet/Allmark/Render/TableCellRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class TableCellRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "table_cell",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderTag.Execute(node, state, "td");
    }
}

File: zig/src/render/tableCellRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderTag = @import("renderTag.zig").renderTag;

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    renderTag(node, state, "td", true);
}

pub const tableCellRenderer = Renderer{
    .name = "table_cell",
    .render = render,
};
